src/sysroot.rs

```diff
@@ -344,6 +364,14 @@ pub struct Stage {
     patch: Table,
 }
 
+/// Which mode to invoke `cargo` in when building the sysroot
+/// Can be either `cargo build` or `cargo check`
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum CargoMode {
```

Comment:
I think `XargoMode` or `SysrootMode` might be a better name for this... because this is no the mode used for the project build, right, just the sysroot?